server/functions/fileSystemFunctions.js

```diff
@@ -247,6 +280,23 @@ async function addFunctionString(applicationId, filename, code) {
     await waitForNewFilesToBeRead()
 }
 
+async function addDependencies(applicationId, newDependencies, isDev) {
+    const packageString = `${newDependencies}${isDev ? '--dev' : ''}`;
+    
+    const appDirectory = `${config.functionsRoot}${applicationId}`;
+
+    execSync(`yarn add ${packageString}`, {cwd: appDirectory});
```

Comment:
What would happen if the package can't be found or doesn't exist?

Reply:
Blows up under the hood with no message to the user :). We should add that, either in vanilla JS/HTML or as part of a react rewrite

Reply:
Updated to at least send back a 500